data/fileio/CSVStreamWriter.h

```diff
@@ -0,0 +1,100 @@
+/* -*- c-basic-offset: 4 indent-tabs-mode: nil -*-  vi:set ts=8 sts=4 sw=4: */
+
+/*
+    Sonic Visualiser
+    An audio file viewer and annotation editor.
+    Centre for Digital Music, Queen Mary, University of London.
+    This file copyright 2017 Lucas Thompson.
+    
+    This program is free software; you can redistribute it and/or
+    modify it under the terms of the GNU General Public License as
+    published by the Free Software Foundation; either version 2 of the
+    License, or (at your option) any later version.  See the file
+    COPYING included with this distribution for more information.
+*/
+
+#ifndef _CSV_STREAM_WRITER_H_
+#define _CSV_STREAM_WRITER_H_
+
+#include "base/BaseTypes.h"
+#include "base/Selection.h"
+#include "base/ProgressReporter.h"
+#include "base/DataExportOptions.h"
+#include "data/model/Model.h"
+#include <QString>
+#include <algorithm>
+
+namespace CSV
+{
```

Comment:
CSV seems a bit of a generic name unless, I guess, you are actually expecting to re-open the namespace for other things. I would consider a more specific name (same as the filename maybe) or even a class with just public static members. Mind you I did get my start writing C++ before namespaces existed.

Reply:
One practical reason to use a more specific name: stack traces. I just tested the code with this change in svcore but without the change in the SV GUI that was needed to take advantage of it (i.e. still using the old API). It crashes because of the uninitialised progress reporter mentioned above, but the first line in the stack trace is not what you might hope to see:

```
#0  0x00005555556adfcc in bool CSV::writeToStreamInChunks<QTextStream>(QTextStream&, Model const&, Selection const&, ProgressReporter*, QString, int, long) ()
#1  0x00005555556adbff in CSVFileWriter::writeSelection(MultiSelection*) ()
#2  0x00005555556ad8eb in CSVFileWriter::write() ()
#3  0x00005555558fd01a in MainWindow::exportAudio(bool) ()
```
(This is a release build, so no file / line numbers etc. So it's useful here if the class name matches the file name.)